<h1>The battle of the neighborhoods - Week 2</h1>

<h2>Introduction and commercial problem:</h2>

<body><font size="5">The city of Bogotá, the capital district of Colombia. It is the center of business, economy and financial capital of Colombia. It is multicultural. It provides many business opportunities and a favorable business environment. It has attracted many players from different markets. It is a global center of business and commerce. The city is an important center of banking and finance, retail, world trade.
This also means that the market is highly competitive. The cost of doing business is also one of the highest. Therefore, any new company or commercial expansion must be analyzed. The knowledge derived from the analysis will provide a good understanding of the business environment that will help to apply strategically to the market. This will help in risk reduction. And the return on investment will be reasonable.</body>

<h2>Description of the problem:</h2>

<body><font size="5">A restaurant is a business that prepares and serves food and drinks to customers in exchange for money, whether paid before the meal, after the meal or with an open account. The city of Bogotá is famous for its excellent cuisine. Its food culture includes a variety of international and national cuisines influenced by the city's immigrant history.Eastern immigrants, Italians, people from the Caribbean, Pacific and eastern plains of the country which have their own typical dishes and recipes.
<ol><font size="5">
<li>Italian immigrants: Bogota-style pizza and Italian cuisine</li>
<li>Immigrants from areas of the country: Paisa tray, shrimp in sauce, boarded, ajiaco.</li>
<li>Chinese restaurants and other Asian restaurants, sandwich restaurants, trattorias</li>
<li>restaurants and coffee shops are everywhere throughout the city.</li>
</ol></body>

<body><font size="5"><p align= center*left*right*justify>It is famous not only for pizzerias, cafes, but also for the excellent restaurants with stars such as Xagal, La Martina, Andrés Carne de Res. The city is home to "almost a thousand of the best and most diverse haute cuisine restaurants in the world" , according to newspaper "El Espectador."
Therefore, it is clear that to survive in such a competitive market it is very important strategically strategic. It is necessary to study several factors to decide the location, stories such as:</body>

<body><font size="5"><p align= center*left*right*justify>It is famous not only for pizzerias, cafes, but also for the excellent restaurants with stars such as Xagal, La Martina, Andrés Carne de Res. The city is home to "almost a thousand of the best and most diverse haute cuisine restaurants in the world" , according to newspaper "El Espectador."</p>
Therefore, it is clear that to survive in such a competitive market it is very important strategically strategic. It is necessary to study several factors to decide the location, stories such as:</body>

<font size="5">
<ul>
<li>Bogota population</li>
<li>Demography of the city of Bogotá</li>
<li>How is the business center of Bogotá demographically constituted?</li>
<li>Are there places like convention centers, offices, business centers?</li>
<li>Who are the competitors in that location?</li>
<li>City segmentation</li>
<li>Untapped markets Saturated Markets, etc.</li>
</ul>

<body><font size="5">The Crepes and Waffles restaurant company. He has decided to open a new store to expand his market in the places of greater affluence of people of the executive or business environment, therefore, he needs to choose the correct location to install his new franchise, if he is successful, they can replicate the same in other situations . For this reason he has decided to invest in a technological solution that will allow him to know the best location of his next restaurant.</body>

<h1>Methodology</h1>

<body><font size="5"><p align= center*left*right*justify>The Methodology area will portray the fundamental parts of our investigation and predication framework. The Methodology segment includes four phases:></p></body>

<font size="5">
<ol>
<li>Collecting Inspection Data</li>
<li>Exploring and Understanding Data</li>
<li>Data preparation and preprocessing</li>
<li>Analysis Data</li>
<li>Description of the best option</li>
</ol>

<h2>Collecting Inspection Data</h2>

<h2>Data 1:</h2>

<body><font size="5"><p align= center*left*right*justify>The capital district of Bogotá has 19 locations. To separate the locations and explore them, we will essentially need a set of data that contains the 19 locations that exist in the district, as well as the latitude and latitude coordinates of each neighborhood.
This set of data exists for free on the web.</p> The link to the data set is <a href="https://es.wikipedia.org/wiki/Anex:Bogot%C3%A1_Localities">https://es.wikipedia.org/wiki/Anex:Bogot%C3%A1_Localities</a> </body>

In [22]:
conda install -c anaconda beautifulsoup4

Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.11
  latest version: 4.8.0

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/jupyterlab/conda/envs/python

  added / updated specs: 
    - beautifulsoup4


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.11.28         |           py36_0         156 KB  anaconda

The following packages will be UPDATED:

    certifi: 2019.11.28-py36_0 conda-forge --> 2019.11.28-py36_0 anaconda
    openssl: 1.1.1d-h516909a_0 conda-forge --> 1.1.1-h7b6447c_0  anaconda


certifi-2019.11.28   | 156 KB    | ##################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done

Note: you may need to restart the kernel to use updated packages.


In [23]:
!conda install -c conda-forge geopy --yes

Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.11
  latest version: 4.8.0

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/jupyterlab/conda/envs/python

  added / updated specs: 
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.11.28         |           py36_0         149 KB  conda-forge

The following packages will be UPDATED:

    certifi: 2019.11.28-py36_0 anaconda --> 2019.11.28-py36_0 conda-forge

The following packages will be DOWNGRADED:

    openssl: 1.1.1-h7b6447c_0  anaconda --> 1.1.1d-h516909a_0 conda-forge


certifi-2019.11.28   | 149 KB    | ##################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done


In [24]:
!conda install -c conda-forge folium=0.5.0 --yes

Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.11
  latest version: 4.8.0

Please update conda by running

    $ conda update -n base -c defaults conda



# All requested packages already installed.



In [25]:
import pandas as pd
import requests
from bs4 import BeautifulSoup

import numpy as np # library to handle data in a vectorized manner

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

#=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

Libraries imported.


In [26]:
website_url = requests.get("https://es.wikipedia.org/wiki/Anexo:Localidades_de_Bogot%C3%A1").text

In [27]:
soup = BeautifulSoup(website_url,'html5lib')
My_table = soup.find('table',{'class':'sortable wikitable'})
My_table
links = My_table.findAll('a')
links
Localidades = []
for link in links:
    Localidades.append(link.get('title'))
    
print(Localidades)
del Localidades[0:2]
Localidades

[None, None, 'Usaquén', 'Chapinero', 'Santa Fe (Bogotá)', 'San Cristóbal (Bogotá)', 'Usme', 'Tunjuelito', 'Bosa (Bogotá)', 'Kennedy (Bogotá)', 'Fontibón', 'Engativá', 'Suba', 'Barrios Unidos (Bogotá)', 'Teusaquillo', 'Los Mártires', 'Antonio Nariño (Bogotá)', 'Puente Aranda', 'La Candelaria', 'Rafael Uribe Uribe (Bogotá)', 'Ciudad Bolívar (Bogotá)', 'Sumapaz (Bogotá)']


['Usaquén',
 'Chapinero',
 'Santa Fe (Bogotá)',
 'San Cristóbal (Bogotá)',
 'Usme',
 'Tunjuelito',
 'Bosa (Bogotá)',
 'Kennedy (Bogotá)',
 'Fontibón',
 'Engativá',
 'Suba',
 'Barrios Unidos (Bogotá)',
 'Teusaquillo',
 'Los Mártires',
 'Antonio Nariño (Bogotá)',
 'Puente Aranda',
 'La Candelaria',
 'Rafael Uribe Uribe (Bogotá)',
 'Ciudad Bolívar (Bogotá)',
 'Sumapaz (Bogotá)']

In [7]:
df = pd.DataFrame()
df['Localidades'] = Localidades
df

Localidades
0                       Usaquén
1                     Chapinero
2             Santa Fe (Bogotá)
3        San Cristóbal (Bogotá)
4                          Usme
5                    Tunjuelito
6                 Bosa (Bogotá)
7              Kennedy (Bogotá)
8                      Fontibón
9                      Engativá
10                         Suba
11      Barrios Unidos (Bogotá)
12                  Teusaquillo
13                 Los Mártires
14      Antonio Nariño (Bogotá)
15                Puente Aranda
16                La Candelaria
17  Rafael Uribe Uribe (Bogotá)
18      Ciudad Bolívar (Bogotá)
19             Sumapaz (Bogotá)

<h1>Information dataframe of the towns of Bogota with its latitude and longitude</h1>

In [28]:
def get_coords_local(localidad, output_as='center'):
    """
    get the bounding box of a locality in WGS84 given its name
    Parameters
    ----------
    localidad : str
        name of the country in english and lowercase
    output_as : 'str
        chose from 'boundingbox' or 'center'. 
         - 'boundingbox' for [latmin, latmax, lonmin, lonmax]
         - 'center' for [latcenter, loncenter]
    Returns
    -------
    output : list
        list with coordinates as str
    """
    # create url
    url = '{0}{1}{2}'.format('http://nominatim.openstreetmap.org/search.php?q=',
                             localidad+', Bogota, Bogota Capital District',
                             '&format=json&polygon=0')
    response = requests.get(url).json()[0]

    # parse response to list
    if output_as == 'boundingbox':
        lst = response[output_as]
        output = [float(i) for i in lst]
    if output_as == 'center':
        lst = [response.get(key) for key in ['lat','lon']]
        output = [float(i) for i in lst]
    return output

In [29]:
df2 = df.copy()

latitudeCln = []
longitudeCln = []
for index, row in df2.iterrows():
    print(row[0])
    lat, long = get_coords_local(localidad=row[0], output_as='center')
    latitudeCln.append(lat)
    longitudeCln.append(long)

df2['Latitude'] = latitudeCln
df2['Longitude'] = longitudeCln
df2

Usaquén
Chapinero
Santa Fe (Bogotá)
San Cristóbal (Bogotá)
Usme
Tunjuelito
Bosa (Bogotá)
Kennedy (Bogotá)
Fontibón
Engativá
Suba
Barrios Unidos (Bogotá)
Teusaquillo
Los Mártires
Antonio Nariño (Bogotá)
Puente Aranda
La Candelaria
Rafael Uribe Uribe (Bogotá)
Ciudad Bolívar (Bogotá)
Sumapaz (Bogotá)


Localidades  Latitude  Longitude
0                       Usaquén  4.694969 -74.031093
1                     Chapinero  4.645377 -74.061943
2             Santa Fe (Bogotá)  4.602204 -74.078837
3        San Cristóbal (Bogotá)  4.548658 -74.047473
4                          Usme  4.508155 -74.114328
5                    Tunjuelito  4.562204 -74.127647
6                 Bosa (Bogotá)  4.609913 -74.184710
7              Kennedy (Bogotá)  4.631582 -74.151319
8                      Fontibón  4.673327 -74.144732
9                      Engativá  4.708695 -74.109643
10                         Suba  4.746565 -74.095228
11      Barrios Unidos (Bogotá)  4.669091 -74.072251
12                  Teusaquillo  4.646410 -74.085441
13                 Los Mártires  4.608375 -74.086538
14      Antonio Nariño (Bogotá)  4.589429 -74.099568
15                Puente Aranda  4.616071 -74.105433
16                La Candelaria  4.596515 -74.073492
17  Rafael Uribe Uribe (Bogotá)  4.573354 -74.122064
18      Ciudad Bolívar (Bogotá)  4.568230 -74.154055
19             Sumapaz (Bogotá)  4.219417 -74.159785

In [30]:
address = 'Bogotá, Colombia'

geolocator = Nominatim(user_agent="capstoneProject")
location = geolocator.geocode(address, timeout=60, exactly_one=True)
latitude = location.latitude
longitude = location.longitude
print('The decimal coordinates of Bogotá are {}, {}.'.format(latitude, longitude))
# make a copy of the dataframe to get it simply back if needed
df3 = df2.copy()
df3

The decimal coordinates of Bogotá are 4.59808, -74.0760439.


Localidades  Latitude  Longitude
0                       Usaquén  4.694969 -74.031093
1                     Chapinero  4.645377 -74.061943
2             Santa Fe (Bogotá)  4.602204 -74.078837
3        San Cristóbal (Bogotá)  4.548658 -74.047473
4                          Usme  4.508155 -74.114328
5                    Tunjuelito  4.562204 -74.127647
6                 Bosa (Bogotá)  4.609913 -74.184710
7              Kennedy (Bogotá)  4.631582 -74.151319
8                      Fontibón  4.673327 -74.144732
9                      Engativá  4.708695 -74.109643
10                         Suba  4.746565 -74.095228
11      Barrios Unidos (Bogotá)  4.669091 -74.072251
12                  Teusaquillo  4.646410 -74.085441
13                 Los Mártires  4.608375 -74.086538
14      Antonio Nariño (Bogotá)  4.589429 -74.099568
15                Puente Aranda  4.616071 -74.105433
16                La Candelaria  4.596515 -74.073492
17  Rafael Uribe Uribe (Bogotá)  4.573354 -74.122064
18      Ciudad Bolívar (Bogotá)  4.568230 -74.154055
19             Sumapaz (Bogotá)  4.219417 -74.159785

<h1>Map of locations in Bogotá</h1>

In [31]:
# create map of Bogotá using latitude and longitude values
map_bogota = folium.Map(location=[latitude, longitude], zoom_start=12)
# add markers to map
for lat, lng, local in zip(df3['Latitude'], df3['Longitude'], df3['Localidades']):
    label = '{}'.format(local)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7).add_to(map_bogota)  
    
map_bogota

In [32]:
# function to repeat the exploring process to all the neighborhoods in Toronto
import urllib
def getNearbyVenues(names, latitudes, longitudes, radius=5000, categoryIds=''):
    try:
        venues_list=[]
        for name, lat, lng in zip(names, latitudes, longitudes):
            #print(name)

            # create the API request URL
            url = 'https://api.foursquare.com/v2/venues/search?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, VERSION, lat, lng, radius, limit)

            if (categoryIds != ''):
                url = url + '&categoryId={}'
                url = url.format(categoryIds)

            # make the GET request
            response = requests.get(url).json()
            results = response["response"]['venues']

            # return only relevant information for each nearby venue
            for v in results:
                success = False
                try:
                    category = v['categories'][0]['name']
                    success = True
                except:
                    pass

                if success:
                    venues_list.append([(
                        name, 
                        lat, 
                        lng, 
                        v['name'], 
                        v['location']['lat'], 
                        v['location']['lng'],
                        v['categories'][0]['name']
                    )])

        nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
        nearby_venues.columns = ['Localidad', 
                  'Localidad Latitude', 
                  'Localidad Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    except:
        print(url)
        print(response)
        print(results)
        print(nearby_venues)

    return(nearby_venues)

In [33]:
limit = 500 # limit of number of venues returned by Foursquare API
radius = 5000 # define radiu
CLIENT_ID = 'UZVWVNFW1EYCXTWMP3BNUTXYXFC1ZCIXWPF4440JVP4OKNP3'
CLIENT_SECRET = 'LYZXKMREOJOQ1A2QYLW1O25IJ1RCUOT0GD4IJEKJVYNMR2S5'
VERSION = '20181020'

<h1>Restaurant dates</h1>

<body><font size="5">
Information regarding the restaurants located in Bogota is sought to determine the amount of restaurant, the type of restaurant, which will allow later to analyze where the location of the new restaurant. Use category id 4bf58dd8d48988d14e941735 to only get the American Restaurant</body>

In [34]:
# Use category id 4bf58dd8d48988d14e941735 to only get the American Restaurant
bogota_venues_Restaurant = getNearbyVenues(names=df3['Localidades'], latitudes=df3['Latitude'], longitudes=df3['Longitude'], radius=1000, categoryIds='4bf58dd8d48988d14e941735')
bogota_venues_Restaurant.head()

Localidad  Localidad Latitude  Localidad Longitude              Venue  \
0   Usaquén            4.694969           -74.031093  La Hamburguesería   
1   Usaquén            4.694969           -74.031093        Mister Ribs   
2   Usaquén            4.694969           -74.031093       Rock 'N Stop   
3   Usaquén            4.694969           -74.031093            Madison   
4   Usaquén            4.694969           -74.031093            Enfuego   

   Venue Latitude  Venue Longitude           Venue Category  
0        4.695310       -74.031716             Burger Joint  
1        4.695403       -74.030600      American Restaurant  
2        4.694666       -74.032456      American Restaurant  
3        4.695443       -74.030262             Burger Joint  
4        4.694105       -74.031857  New American Restaurant

<body><font size="5">
Get a total of 74 restaurants in Bogota that is defined as the competition of the company with its latitude and longitude</body>

In [35]:
bogota_venues_Restaurant.shape

(74, 7)

In [36]:
# function to add markers for given venues to map
def addToMap(df, color, existingMap):
    for lat, lng, local, venue, venueCat in zip(df['Venue Latitude'], df['Venue Longitude'], df['Localidad'], df['Venue'], df['Venue Category']):
        label = '{} ({}) - {}'.format(venue, venueCat, local)
        label = folium.Popup(label, parse_html=True)
        folium.CircleMarker(
            [lat, lng],
            radius=5,
            popup=label,
            color=color,
            fill=True,
            fill_color=color,
            fill_opacity=0.7).add_to(existingMap)

<h1>Map Restaurant location</h1>

In [37]:
map_bogota_Restaurant = folium.Map(location=[latitude, longitude], zoom_start=12)
addToMap(bogota_venues_Restaurant, 'red', map_bogota_Restaurant)
map_bogota_Restaurant

<h1>Business Center Dates </h1>
<body><font size="5">
Information regarding the Business Center located in Bogota is sought to determine the amount of Business Center, which will allow later to analyze where the location of the new restaurant. Use category id 56aa371be4b08b9a8d573517 to only get the Bussiness Center</body>

In [38]:
# Use category id 56aa371be4b08b9a8d573517 to only get the Bussiness Center
bogota_venues_BusinessCenter = getNearbyVenues(names=df3['Localidades'], latitudes=df3['Latitude'], longitudes=df3['Longitude'], radius=1000, categoryIds='56aa371be4b08b9a8d573517')
bogota_venues_BusinessCenter.head()

Localidad  Localidad Latitude  Localidad Longitude  \
0   Usaquén            4.694969           -74.031093   
1   Usaquén            4.694969           -74.031093   
2   Usaquén            4.694969           -74.031093   
3   Usaquén            4.694969           -74.031093   
4   Usaquén            4.694969           -74.031093   

                           Venue  Venue Latitude  Venue Longitude  \
0                     Flormorado        4.694266       -74.032539   
1                  Usaquén Plaza        4.696983       -74.031150   
2       Kaiwa Centro Empresarial        4.698918       -74.030767   
3  Edwards Lifesciences Colombia        4.692951       -74.033920   
4           Ignition Agencia BTL        4.699775       -74.032768   

    Venue Category  
0  Business Center  
1    Shopping Mall  
2  Business Center  
3  Business Center  
4  Business Center

<body><font size="5">
Get a total of 27 Business Center in Bogota that is defined as the competition of the company with its latitude and longitude</body>

In [39]:
bogota_venues_BusinessCenter.shape

(27, 7)

<h1>Map Business Center location</h1>

In [40]:
map_bogota_venues_BusinessCenter= folium.Map(location=[latitude, longitude], zoom_start=12)
addToMap(bogota_venues_BusinessCenter,'green', map_bogota_venues_BusinessCenter)
map_bogota_venues_BusinessCenter


<h1>Convention Center Dates </h1>
<body><font size="5">
Information regarding the Convention Center located in Bogota is sought to determine the amount of Convention Center, which will allow later to analyze where the location of the new restaurant. Use category id 4bf58dd8d48988d1ff931735 to only get the Convention Center</body>

In [41]:
# Use category id 4bf58dd8d48988d1ff931735 to only get the Convention Center
bogota_venues_concenter = getNearbyVenues(names=df3['Localidades'], latitudes=df3['Latitude'], longitudes=df3['Longitude'], radius=1000, categoryIds='4bf58dd8d48988d1ff931735')
bogota_venues_concenter.head()

Localidad  Localidad Latitude  Localidad Longitude  \
0   Usaquén            4.694969           -74.031093   
1   Usaquén            4.694969           -74.031093   
2   Usaquén            4.694969           -74.031093   
3   Usaquén            4.694969           -74.031093   
4   Usaquén            4.694969           -74.031093   

                                         Venue  Venue Latitude  \
0                Centro Empresarial Paseo Real        4.697782   
1                                           AR        4.692235   
2                    AR-centro de convenciones        4.692206   
3                 Villa Santa Maria De Usaquen        4.695387   
4  1ra Convención Crowdfunding Prodigy Network        4.693362   

   Venue Longitude     Venue Category  
0       -74.029324  Convention Center  
1       -74.035281  Convention Center  
2       -74.035335  Convention Center  
3       -74.031345       Meeting Room  
4       -74.033349  Convention Center

<body><font size="5">
Get a total of 41 Convention Center in Bogota that is defined as the competition of the company with its latitude and longitude</body>

In [42]:
bogota_venues_concenter.shape

(41, 7)

<h1>Map Convention Center location</h1>

In [43]:
map_bogota_concenter = folium.Map(location=[latitude, longitude], zoom_start=12)
addToMap(bogota_venues_concenter, 'gold', map_bogota_concenter)
map_bogota_concenter

<h1>Office Dates </h1>
<body><font size="5">
Information regarding the office located in Bogota is sought to determine the amount of Convention Center, which will allow later to analyze where the location of the new restaurant. Use category id 4d4b7105d754a06375d81259 to only get the office</body>

In [44]:
# Use category id 4d4b7105d754a06375d81259 to only get the office
bogota_venues_office = getNearbyVenues(names=df3['Localidades'], latitudes=df3['Latitude'], longitudes=df3['Longitude'], radius=1000, categoryIds='4d4b7105d754a06375d81259')
bogota_venues_office.head()

Localidad  Localidad Latitude  Localidad Longitude  \
0   Usaquén            4.694969           -74.031093   
1   Usaquén            4.694969           -74.031093   
2   Usaquén            4.694969           -74.031093   
3   Usaquén            4.694969           -74.031093   
4   Usaquén            4.694969           -74.031093   

                        Venue  Venue Latitude  Venue Longitude  \
0                  Bistronomy        4.695679       -74.030267   
1  Centro Médico de la Sabana        4.696026       -74.031596   
2    Santa Ana Medical Center        4.696085       -74.032065   
3         Amazon Web Services        4.694108       -74.032392   
4               Usaquén Plaza        4.696983       -74.031150   

      Venue Category  
0  French Restaurant  
1    Doctor's Office  
2    Doctor's Office  
3             Office  
4      Shopping Mall

<body><font size="5">
Get a total of 792 office in Bogota that is defined as the competition of the company with its latitude and longitude</body>

In [45]:
bogota_venues_office.shape

(793, 7)

<h1>Map office location</h1>

In [46]:
map_bogota_office = folium.Map(location=[latitude, longitude], zoom_start=12)
addToMap(bogota_venues_office, 'fuchsia', map_bogota_office)
map_bogota_office

In [47]:
def addColumn(startDf, columnTitle, dataDf):
    grouped = dataDf.groupby('Localidad').count()
    
    for n in startDf['Localidad']:
        try:
            startDf.loc[startDf['Localidad'] == n,columnTitle] = grouped.loc[n, 'Venue']
        except:
            startDf.loc[startDf['Localidad'] == n,columnTitle] = 0

<h1>Category number by location </h1>
<body><font size="5">The number per category corresponding to each location was found. with this, weighting the categories and establishing the best location for the next Crepes and Waffles restaurant</body>

In [48]:
df_data = df3.copy()
df_data.rename(columns={'Localidades':'Localidad'}, inplace=True)
addColumn(df_data, 'Restaurant', bogota_venues_Restaurant)
addColumn(df_data, 'Business Center', bogota_venues_BusinessCenter)
addColumn(df_data, 'Convention Center', bogota_venues_concenter)
addColumn(df_data, 'Offices', bogota_venues_office)
df_data

Localidad  Latitude  Longitude  Restaurant  \
0                       Usaquén  4.694969 -74.031093        12.0   
1                     Chapinero  4.645377 -74.061943        15.0   
2             Santa Fe (Bogotá)  4.602204 -74.078837        15.0   
3        San Cristóbal (Bogotá)  4.548658 -74.047473         0.0   
4                          Usme  4.508155 -74.114328         0.0   
5                    Tunjuelito  4.562204 -74.127647         0.0   
6                 Bosa (Bogotá)  4.609913 -74.184710         0.0   
7              Kennedy (Bogotá)  4.631582 -74.151319         3.0   
8                      Fontibón  4.673327 -74.144732         1.0   
9                      Engativá  4.708695 -74.109643         2.0   
10                         Suba  4.746565 -74.095228         5.0   
11      Barrios Unidos (Bogotá)  4.669091 -74.072251         0.0   
12                  Teusaquillo  4.646410 -74.085441         1.0   
13                 Los Mártires  4.608375 -74.086538         3.0   
14      Antonio Nariño (Bogotá)  4.589429 -74.099568         1.0   
15                Puente Aranda  4.616071 -74.105433         0.0   
16                La Candelaria  4.596515 -74.073492        16.0   
17  Rafael Uribe Uribe (Bogotá)  4.573354 -74.122064         0.0   
18      Ciudad Bolívar (Bogotá)  4.568230 -74.154055         0.0   
19             Sumapaz (Bogotá)  4.219417 -74.159785         0.0   

    Business Center  Convention Center  Offices  
0               8.0                8.0     50.0  
1               6.0                5.0     50.0  
2               1.0                6.0     50.0  
3               0.0                0.0      0.0  
4               0.0                0.0     31.0  
5               0.0                0.0     44.0  
6               0.0                2.0     39.0  
7               0.0                1.0     43.0  
8               3.0                2.0     44.0  
9               0.0                1.0     48.0  
10              2.0                1.0     47.0  
11              3.0                1.0     49.0  
12              0.0                4.0     47.0  
13              1.0                1.0     43.0  
14              2.0                2.0     45.0  
15              0.0                2.0     47.0  
16              1.0                5.0     50.0  
17              0.0                0.0     44.0  
18              0.0                0.0     22.0  
19              0.0                0.0      0.0

In [49]:
# negative weight, because Jeronimo wants to open a burger joint and thus wants to avoid concurrence as much as possible
weight_Restaurant = -1

# positive weight, because high school students are good customers
weight_BusinessCenter = 1

# positive weight, because uni students are good customers
weight_concenter = 1.5

# positive weight because employees are even better customers
weight_offices = 2

<h1>weighting of the analis categories </h1>
<body><font size="5">As stated above, the categories will be weighted to assign a score and thus, know which is the best location and the location where the restaurant should be.</body>

In [50]:
df_weighted = df_data[['Localidad']].copy()

In [53]:
df_weighted['Score'] = df_data['Restaurant'] * weight_Restaurant + df_data['Business Center'] * weight_BusinessCenter + df_data['Convention Center'] * weight_concenter + df_data['Offices'] * weight_offices
df_weighted = df_weighted.sort_values(by=['Score'], ascending=False)
df_weighted

Localidad  Score
0                       Usaquén  108.0
11      Barrios Unidos (Bogotá)  102.5
12                  Teusaquillo   99.0
1                     Chapinero   98.5
15                Puente Aranda   97.0
9                      Engativá   95.5
2             Santa Fe (Bogotá)   95.0
14      Antonio Nariño (Bogotá)   94.0
8                      Fontibón   93.0
16                La Candelaria   92.5
10                         Suba   92.5
5                    Tunjuelito   88.0
17  Rafael Uribe Uribe (Bogotá)   88.0
13                 Los Mártires   85.5
7              Kennedy (Bogotá)   84.5
6                 Bosa (Bogotá)   81.0
4                          Usme   62.0
18      Ciudad Bolívar (Bogotá)   44.0
3        San Cristóbal (Bogotá)    0.0
19             Sumapaz (Bogotá)    0.0

<H1>Results</H1>

<body><font size="5">When obtaining the score it is observed that the town of Usaquen is the best location in the district of Bogota when obtaining 108 points and the second neighborhoods united when obtaining 102 points.</body>

<h1>Map best Location of restaurant </h1>
<body><font size="5">On the map you can see the best location found by the system.</body>

In [54]:
map_bog_result = folium.Map(location=[latitude, longitude], zoom_start=15)

bog_win = df3[df3['Localidades'] == 'La Candelaria']

for lat, lng, local in zip(bog_win['Latitude'], bog_win['Longitude'], bog_win['Localidades']):
    label = '{}'.format(local)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='blue',
        fill_opacity=0.7).add_to(map_bog_result) 

addToMap(bogota_venues_Restaurant[bogota_venues_Restaurant['Localidad'] == 'Usaquén'], 'red', map_bog_result)
addToMap(bogota_venues_BusinessCenter[bogota_venues_BusinessCenter['Localidad'] == 'Usaquén'], 'green', map_bog_result)
addToMap(bogota_venues_concenter[bogota_venues_concenter['Localidad'] == 'Usaquén'], 'gold', map_bog_result)
addToMap(bogota_venues_office[bogota_venues_office['Localidad'] == 'Usaquén'], 'fuchsia', map_bog_result)

map_bog_result

<h1>discuss any observations</h1>

<body><font size="5">If you have more information on the preferences of dishes of potential customers, referential information of the places where it usually.

machine learning could be used as a method to further improve the decision for the company to decide the location of its new restaurant, on the other hand, it can be seen that there is not a single option, therefore the company is suggested to open more of a room, taking into account the information provided by the analysis</body>

<h1>Conclusion</h1>

<body><font size="5">The stakeholder's problem is resolved. Stakeholder wants to find the best place to build a new restaurant in Bogota, and the factors of "best location" is based on the number of near restaurants, business centers, Convention Center, office category around the location. Stakeholder also provide his favourite list of place to work to further explain what the "best location" is. Content-based filtering ponderation technique is the most suitable method to resolve the problem. It combines stakeholder's preference in the city and profile to make the recommendation result.</body>